pydantic/fields.py

```diff
@@ -95,6 +96,8 @@ def __init__(self, **kwargs: Any) -> None:
         self.init_var = kwargs.get('init_var', None)
         self.kw_only = kwargs.get('kw_only', None)
         self.validate_default = kwargs.get('validate_default', None)
+        self.frozen = kwargs.get('frozen', None)
+        self.final = kwargs.get('final', None)
```

Comment:
I thought we were combining these?

If there's a good reason not to, fine by me.

Reply:
I think the reason I didn't combine them was two-fold: one, `ModelField` had an attribute called `final` in v1, but not `frozen`. So it felt a little weird to remove `final`, especially considering conceptually it's different from `frozen` (for starters, in theory we shouldn't allow overriding the default, though I suspect it's not worth the effort to enforce outside of type checking).

Second, I imagined we might eventually want to give different error messages depending on frozen vs. final. (I could imagine a newer user getting an error message saying 'frozen' when you just annotated as Final and possibly don't even know what frozen refers to could be confusing.) In which case perhaps it would be helpful to have the distinction.

However, in this PR, I just set the frozen attribute on the field based on `frozen=field_info.frozen or field_info.final`. So yeah maybe we should just combine the two and only store `frozen` on the field.

If you read the above and don't have an opinion/haven't changed your mind, I suppose I'll just combine them and drop the `final` attribute, and it can be a feature request in the future.

Reply:
fine, let's keep it.